### ISSUE TITLE
[strict-boolean-expressions] ignoreRhs doesn’t work

### ISSUE THREAD

**Issue by cevek** [typescript-eslint/typescript-eslint#993]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/strict-boolean-expressions": ["error", { "ignoreRhs": true }]
  }
}
```

```TS
declare const a: number | undefined; 
const x = a || 1;
```

**Expected Result**
no error

**Actual Result**
Unexpected non-boolean in conditional

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.3.1` |
| `@typescript-eslint/parser`        | `2.2` |
| `TypeScript`                       | `3.6` |
| `ESLint`                           | `6.3` |
| `node`                             | `12` |
| `npm`                              | `6` |

**Comment by bradzacher** [comment]
`ignoreRhs` won't let you use non-booleans on the left hand side...
`a` is of type `number | undefined`, which is not a strict boolean.

As mentioned in the docs:

> It is useful to be explicit, for example, if you were trying to check if a number was defined. Doing if (number) would evaluate to false if number was defined and 0. ***This rule forces these expressions to be explicit and to strictly use booleans.***
